### PR TITLE
Add YAML (Jinja) syntax

### PR DESCRIPTION
**Pull request by deathaxe**
This PR suggests to add a YAML (Jinja) syntax, 
which extends ST's core YAML to embed jinja statements and expressions.

It does by intent not use `with_prototype` to be able to extend specific YAML contexts 
in proper ways to maintain meta scoping.

Notes:

1. It creates a dedicated HTML (Jinja), in order to reduce syntax cache size of YAML (Jinja).
2. HTML (Jinja) still uses the "push with_prototype" approach.
   Changing it is out of scope for this PR.

### CODE REVIEW

**Review by FichteFoll — APPROVED** [review]

**Comment by FichteFoll** [comment]
Perhaps the supported syntaxes in the readme should also be updated.

**Comment by deathaxe** [comment]
Maybe part of the next - somewhat larger step of refactoring HTML as it will automatically cause this package to support CSS/JavaScript/JSON/HTML/YAML automatically - with drastically reduced compiled syntax cache sizes.

File: resources/completions/BetterJinja (Filters).sublime-completions [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "scope": "text.jinja text.html.basic & (meta.placeholder.jinja | meta.statement.jinja)",
+    "scope": "(text.jinja, text.html.jinja, source.yaml.jinja) & (meta.interpolation.jinja | meta.statement.jinja)",
```
Comment (FichteFoll) [inline resources/completions/BetterJinja (Filters).sublime-completions:2]: I wonder if defining a meta scope `meta.jinja` and using that in all Jinja-enabled syntaxes would simplify this a lot, especially if more are going to be added.

**Review by FichteFoll — APPROVED** [review]

File: resources/completions/BetterJinja (Filters).sublime-completions (hunk 1 above)
Comment (deathaxe) [reply resources/completions/BetterJinja (Filters).sublime-completions:2]: Sounds useful. Actually something we probably should consider for bundled syntaxes as well.

Snippets/Completions selectors is a common issue, which I haven't paid much attention before.

**Comment by nicholas-ochoa** [comment]
Any chance of adding support for Jinja statements or comments? From the Jinja docs:

```
There are a few kinds of delimiters. The default Jinja delimiters are configured as follows:

{% ... %} for [Statements](https://jinja.palletsprojects.com/en/latest/templates/#list-of-control-structures)

{{ ... }} for [Expressions](https://jinja.palletsprojects.com/en/latest/templates/#expressions) to print to the template output

{# ... #} for [Comments](https://jinja.palletsprojects.com/en/latest/templates/#comments) not included in the template output
```

https://jinja.palletsprojects.com/en/latest/templates/

**Comment by deathaxe** [comment]
They already are.

**Comment by deathaxe** [comment]
Closing in favor of https://github.com/Sublime-Instincts/BetterJinja/pull/22

**Pull request closed without merging**